.github/workflows/mcp_unit_test.yml

```diff
@@ -0,0 +1,65 @@
+name: 'MCP Unit Tests'
+
+on:
+  pull_request:
+    branches:
+      - master
```

Comment:
Have a look at github.base_ref as a condition for pull requests to master

Reply:
The if statement within the job currently does that, and it doesn't seem possible to do this within the on section.
[`if: contains(github.head_ref, 'MCP') == true` ](https://github.com/mesoform/terraform-infrastructure-modules/blob/b40e1940d266222039a6a1b37783b3b7b92bdef6/.github/workflows/mcp_unit_test.yml#L15)
Will only run the job if the source branch contains "MCP"; meaning the workflow will be marked as skipped, for branches that aren't MCP related, in the github actions list. And the on section means the conditional will only be evaluated on pull requests, and pushes to master.
Don't think it is possible have conditionals within the on section, possible keys are `branches`, `branches-ignore`, `tags`, `tags-ignore`, `paths`, `paths-ignore` and `type`

Reply:
My point here is to look at a different property 

> The base_ref or target branch of the pull request in a workflow run. This property is only available when the event that triggers a workflow run is a pull_request.